packages/java/endpoint/src/main/java/com/vaadin/hilla/startup/RouteUnifyingServiceInitListener.java

```diff
@@ -60,10 +67,14 @@ public class RouteUnifyingServiceInitListener
     @Autowired
     public RouteUnifyingServiceInitListener(
             ClientRouteRegistry clientRouteRegistry, RouteUtil routeUtil,
-            RouteUnifyingConfigurationProperties routeUnifyingConfigurationProperties) {
+            RouteUnifyingConfigurationProperties routeUnifyingConfigurationProperties,
+            @Nullable NavigationAccessControl accessControl,
+            @Nullable ViewAccessChecker viewAccessChecker) {
```

Comment:
This was probably discussed already. For non-Spring apps, how the access control `BeforeEnterListener` instances are used and passed to the Menu-related logic, as they aren't auto-registered? 

Well, if the Spring-Boot not used, Hilla isn't used and Hilla main layout isn't there anyway, so maybe it's not that important.

Reply:
I understand the same, when there's no Hilla in the picture.

Reply:
Listeners should be registered in `UIInternals`, so that's where they can be read like this:
```
UI.getCurrent().getInternals()
  .getListeners(BeforeEnterHandler.class).stream()
  .filter(handler -> handler instanceof BeforeEnterListener)
  .map(BeforeEnterListener.class::cast).toList()
```